src/components/PostsList.tsx

```diff
@@ -3,52 +3,60 @@
 import classNames from 'classnames';
 import React from 'react';
 import { Post } from '../types/Post';
+import { useAppDispatch } from '../app/hooks';
+import {
+  clearSelectedPost,
+  setSelectedPost,
+} from '../features/selectedPostSlice';
 
 type Props = {
   posts: Post[];
   selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
 };
 
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+export const PostsList: React.FC<Props> = ({ posts, selectedPostId = 0 }) => {
+  const dispatch = useAppDispatch();
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+  const showPosts = posts.map(post => (
+    <tr key={post.id} data-cy="Post">
+      <td data-cy="PostId">{post.id}</td>
+      <td data-cy="PostTitle">{post.title}</td>
+      <td className="has-text-right is-vcentered">
+        <button
+          type="button"
+          data-cy="PostButton"
+          className={classNames('button', 'is-link', {
+            'is-light': post.id !== selectedPostId,
+          })}
+          onClick={() => {
+            dispatch(
+              post.id === selectedPostId
+                ? clearSelectedPost()
+                : setSelectedPost(post),
+            );
+          }}
+        >
+          {post.id === selectedPostId ? 'Close' : 'Open'}
```

Comment:
Consider using more descriptive text for the button label instead of 'Close' and 'Open'. This can improve accessibility and user experience by clearly indicating the action that will be performed.